.github/workflows/conda.yml

```diff
@@ -38,6 +54,15 @@ jobs:
           use-mamba: true
           python-version: 3.9
           channel-priority: strict
+      - name: Remove outdated conda nightlies
+        if: github.event_name == 'push' && github.repository == 'dask/dask' && startsWith(github.ref, 'refs/tags/')
+        env:
+          ANACONDA_API_TOKEN: ${{ secrets.DASK_CONDA_TOKEN }}
+        run: |
+          # install anaconda for removal
+          mamba install -c conda-forge anaconda-client
+
+          anaconda remove dask/dask-core --force
```

Comment:
Do we want to make sure the nightlies removed are older than a certain version? Or is that case unlikely to occur?

Reply:
Definitely think if we wanted to, we could be explicit here and specify to remove all packages older than the current computed "nightly" version rather than everything, which could guard against the edge case that we somehow upload post-release nighties to the channel _before_ the release workflow run is able to reach this point.

Were you thinking of any particular edge cases? In my head, I see the above case as niche enough that we'd probably not encounter even if we merged this in as is, but want to make sure I'm not missing a more crucial situation here